### DISCUSSION TITLE
Missing optional RelatedField leads to KeyError when accessing Serializer.data

### DISCUSSION THREAD

**Discussion by dtomas** [encode/django-rest-framework#8931] [Potential Issue]
When accessing the `.data` property of a `Serializer` with an optional `RelatedField`, a `KeyError` is raised if the field is missing from `serializer.validated_data`, since the exception is not being caught in [`RelatedField.get_attribute`](https://github.com/encode/django-rest-framework/blob/6b73acc1735631df1e666cc87eee0d14de6ae018/rest_framework/relations.py#L175).

In [`Field.get_attribute`](https://github.com/encode/django-rest-framework/blob/6b73acc1735631df1e666cc87eee0d14de6ae018/rest_framework/fields.py#L470) however, `SkipField` is raised in this case. Shouldn't this be done for optional related fields as well?

**Comment by auvipy** [comment]
can you please share code examples and related docs of what you are trying to achieve?
